pkg/controllers/user/cis/clusterScanHandler.go

```
@@ -0,0 +1,161 @@
+	userCtx                      *config.UserContext
+	userCtxNSClient              rcorev1.NamespaceInterface
+	clusterNamespace             string
+	systemAccountManager         *systemaccount.Manager
+	configMapsClient             rcorev1.ConfigMapInterface
+}
+
+func (csh *cisScanHandler) Create(cs *mgmtv3.ClusterScan) (runtime.Object, error) {
+	logrus.Debugf("cisScanHandler: Create: %+v", cs)
+	var err error
+	cluster, err := csh.mgmtCtxClusterClient.Get(cs.ClusterName, metav1.GetOptions{})
+	if err != nil {
+		return cs, err
+	}
+	if !v3.ClusterConditionReady.IsTrue(cluster) {
+		return cs, fmt.Errorf("cisScanHandler: Create: cluster not ready")
+	}
+	if !mgmtv3.ClusterScanConditionCreated.IsTrue(cs) {
+		logrus.Infof("cisScanHandler: Create: deploying helm chart")
+		// Deploy the system helm chart
+		if err := csh.deployApp(cs.Spec.ClusterID, cs.Name); err != nil {
+			logrus.Errorf("cisScanHandler: Create: error deploying app: %v", err)
+			return cs, err
+		}
+		mgmtv3.ClusterScanConditionCreated.True(cs)
+		mgmtv3.ClusterScanConditionCompleted.Unknown(cs)
+
+		cs, err = csh.mgmtCtxClusterScanClient.Update(cs)
+		if err != nil {
+			return cs, err
+		}
+	}
+	return cs, nil
+}
+
+func (csh *cisScanHandler) Remove(cs *mgmtv3.ClusterScan) (runtime.Object, error) {
+	logrus.Debugf("cisScanHandler: Remove: %+v", cs)
+	// Delete the configmap associated with this scan
+	err := csh.configMapsClient.Delete(cs.Name, &metav1.DeleteOptions{})
+	if err != nil {
+		return cs, fmt.Errorf("cisScanHandler: Delete: error deleting cm=%v", cs.Name)
+	}
+
+	return cs, nil
+}
+
+func (csh *cisScanHandler) Updated(cs *mgmtv3.ClusterScan) (runtime.Object, error) {
+	logrus.Debugf("cisScanHandler: Updated: %+v", cs)
+	if !mgmtv3.ClusterScanConditionCompleted.IsUnknown(cs) &&
+		mgmtv3.ClusterScanConditionCompleted.IsFalse(cs) {
+		// Delete the system helm chart
+		if err := csh.deleteApp(cs.Name); err != nil {
+			logrus.Errorf("cisScanHandler: error deleting app: %v", err)
```

should just return the error. It's better to not log in the methods, and instead return the error back to the framework